extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticationMechanism.java

```diff
@@ -15,4 +16,11 @@
 
     CompletionStage<Boolean> sendChallenge(RoutingContext context);
```

Comment:
Rather than adding these extra methods there should be a ComplectionStage<ChallengeResult> getChallenge(RoutingContext context). The result should contain this extra information. sendChallenge should have a default implementation added that delegates to the results of this method.

Reply:
and then translate the exception in a vertx filter/handler instead of JAX-RS exception mapper?

Reply:
tried that, unfortunately doesn't play well with the undertow exception mapping.
Willadd the `getChallenge` and stay on resteasy exception mapper and read the challenge in a blocking manner for now.

Reply:
I added the first part of the rework and tried to address it. Is this what you meant @stuartwdouglas ?